sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobServiceImpl.java

```diff
@@ -1485,5 +1686,19 @@ T getResult() {
                                            @Header("x-ms-encryption-key") String encryptionKey,
                                            @Header("x-ms-encryption-key-sha256") String encryptionKeySha256,
                                            @Header("x-ms-encryption-algorithm") EncryptionAlgorithmType encryptionAlgorithm);
+
+        @DELETE("{containerName}/{blob}")
+        Call<Void> delete(@Path("containerName") String containerName,
```

Comment:
return type should be `Call <ResponseBody>`

Reply:
I assume we should also do this for APIs like `getBlobProperties` where we don't expect a body in a successful call, but there could be one in a failed call, right?

Reply:
Also, the reason I went with `Void` is because the Java API returns a `Mono<Void>` for the [delete operation](https://github.com/Azure/azure-sdk-for-java/blob/712e35f4ad597d43d34c678cc263dbc9095a3ed6/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobAsyncClientBase.java#L1071) and the [`BlobDeleteResponse`](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/models/BlobsDeleteResponse.java) object.

Reply:
I guess this means we want to return a `Void` at the public API level but a `ResponseBody` at the service interface level. Is this correct?

Reply:
_I assume we should also do this for APIs like getBlobProperties where we don't expect a body in a successful call, but there could be one in a failed call, right?_

Yes, it seems there is a similar bug in `getBlobProperties` as well, that needs to be fixed.

_I guess this means we want to return a Void at the public API level but a ResponseBody at the service interface level. Is this correct?_

right, like java SDK, from public API we return `Void` but in internal service interface we need `ResponseBody` so that we can read error body from it and throw.